utils/clients/client.go

```diff
@@ -59,6 +60,39 @@ func New(serverURL string, authToken string) (*SimpleHTTPClient, error) {
 	}, nil
 }
 
+// NewWithProxy returns a new SimpleHTTPClient, retrieving the base URL from the environment and adds a proxy
+func NewWithProxy(name string, serverURL string, authToken string, proxyURL string) (*SimpleHTTPClient, error) {
+	baseURL, err := url.Parse(serverURL)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var proxy func(*http.Request) (*url.URL, error)
+	if len(proxyURL) != 0 {
+		proxiedURL, err := url.Parse(proxyURL)
+		if err != nil {
+			panic("UPHOLD_HTTP_PROXY is not a valid proxy URL")
```

Comment:
if this is going to be a more general client utility function - perhaps we should just look for `HTTP_PROXY` instead?

Reply:
fixed in 9cd17a8